hivemind_bus_client/encodings/benchmark.py

```
@@ -0,0 +1,289 @@
+    print("=" * 55)
+    for row in table:
+        print(f"{row[0]:<20} {row[1]:<12.2f} {row[2]:<10.2f} {row[3]:<10.2f}")
+
+    if output:
+        save_detailed_results_to_markdown(results, output)
+
+
+@click.command()
+@click.option("--sizes", default="10,100,1000,5000,10000,50000", help="Data sizes to benchmark, comma-separated.")
+@click.option("--weights", default="0.5,0.5", help="Weights for performance and bandwidth, comma-separated.")
+def main(sizes: str, weights: str):
```

Ruff (0.8.2)

Redefinition of unused `main` from line 184

(F811)